airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeDestinationHandler.java

```diff
@@ -72,6 +76,47 @@ public boolean isFinalTableEmpty(final StreamId id) throws SQLException {
     return rowCount == 0;
   }
 
+  @Override
+  public Optional<Instant> getMinTimestampForSync(final StreamId id) throws Exception {
+    final boolean rawTableExists = database.queryInt(
+        """
+        SELECT count(1)
+        FROM information_schema.tables
+        WHERE table_catalog = ?
+          AND table_schema = ?
+          AND table_name = ?
+        """,
+        databaseName.toUpperCase(),
+        id.rawNamespace(),
+        id.rawName()) == 1;
+    if (!rawTableExists) {
+      return Optional.empty();
+    }
+    final List<String> queryResult = database.queryStrings(
+        conn -> conn.createStatement().executeQuery(new StringSubstitutor(Map.of(
+            "raw_table", id.rawTableId(SnowflakeSqlGenerator.QUOTE))).replace(
+                // snowflake timestamps have nanosecond precision
+                """
+                SELECT to_varchar(
+                  COALESCE(
+                    (
+                      SELECT TIMESTAMPADD(NANOSECOND, -1, MIN("_airbyte_extracted_at"))
+                      FROM ${raw_table}
+                      WHERE "_airbyte_loaded_at" IS NULL
+                    ),
+                    (
+                      SELECT MAX("_airbyte_extracted_at")
+                      FROM ${raw_table}
+                    )
+                  ),
+                  'YYYY-MM-DDTHH24:MI:SS.FF9TZH:TZM'
+                ) AS MIN_TIMESTAMP
```

Comment:
Since we are in the realm of performacne - does having the COALESCE mean that we will always exec 2 table scans here (`min(_airbyte_extracted_at)` and `max(_airbyte_extracted_at)`), or will the second sub-query not be executed if the first one is non-null?  If so, we could split this into 2 SQL statements to skip part 2 if part one returns something

Reply:
owait you're suggesting just running two statements in sequence. That sounds easier :P